Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/interpolated_corrected_curvatures.h

```diff
@@ -0,0 +1,1673 @@
+    d_max = (std::max)(d_sq, d_max);
+    d_min = (std::min)(d_sq, d_min);
+  }
+
+  // if the furthest point is inside ball, return 1
+  if (d_max <= r * r) return 1.0;
+  // if the closest point is outside ball, return 0
+  else if (r * r <= d_min) return 0.0;
+
+  // else, approximate inclusion ratio of the triangle:
+  d_max = sqrt(d_max);
+  d_min = sqrt(d_min);
+  return (r - d_min) / (d_max - d_min);
+}
+
+template<typename GT>
+Principal_curvatures_and_directions<GT> principal_curvatures_and_directions_from_anisotropic_measures(
+  const std::array<typename GT::FT, 3 * 3> anisotropic_measure,
+  const typename GT::FT v_mu0,
+  const typename GT::Vector_3 u_GT,
+  const typename GT::FT avg_edge_length
+)
+{
+  // putting anisotropic measure in matrix form
+  Eigen::Matrix<typename GT::FT, 3, 3> v_muXY = Eigen::Matrix<typename GT::FT, 3, 3>::Zero();
+
+  for (std::size_t ix = 0; ix < 3; ix++)
+    for (std::size_t iy = 0; iy < 3; iy++)
+      v_muXY(ix, iy) = anisotropic_measure[ix * 3 + iy];
+
+  // constant factor K to force the principal direction eigenvectors to be tangential to the surface
+  Eigen::Matrix<typename GT::FT, 3, 1> u(u_GT.x(), u_GT.y(), u_GT.z());
+  const typename GT::FT K = 1000 * avg_edge_length * v_mu0;
+
+  // symmetrizing and adding the constant term
+  v_muXY = 0.5 * (v_muXY + v_muXY.transpose()) + K * u * u.transpose();
+
+
+  // computing eigenvalues and eigenvectors
+  Eigen::SelfAdjointEigenSolver<Eigen::Matrix <typename GT::FT, 3, 3>> eigensolver;
+
+  eigensolver.computeDirect(v_muXY);
+
+  if (eigensolver.info() != Eigen::Success)
+    return Principal_curvatures_and_directions<GT>();
+
+  const Eigen::Matrix<typename GT::FT, 3, 1> eig_vals = eigensolver.eigenvalues();
+  const Eigen::Matrix<typename GT::FT, 3, 3> eig_vecs = eigensolver.eigenvectors();
+
+  const typename GT::Vector_3 min_eig_vec(eig_vecs(0, 1), eig_vecs(1, 1), eig_vecs(2, 1));
+  const typename GT::Vector_3 max_eig_vec(eig_vecs(0, 0), eig_vecs(1, 0), eig_vecs(2, 0));
+
+  // returning principal curvatures and directions (with the correct sign)
+  return Principal_curvatures_and_directions<GT>(
+    (!is_zero(v_mu0)) ? -eig_vals[1] / v_mu0 : 0.0,
+    (!is_zero(v_mu0)) ? -eig_vals[0] / v_mu0 : 0.0,
+    min_eig_vec,
+    max_eig_vec
+    );
+}
+
+// measures are computed for faces only if they are adjacent to the vertex
+template<typename GT, typename PolygonMesh, typename VPM, typename VNM>
+Vertex_measures<GT> interpolated_corrected_measures_one_vertex_no_radius(
+  const PolygonMesh& pmesh,
+  const typename boost::graph_traits<PolygonMesh>::vertex_descriptor v,
+  const bool is_mean_curvature_selected,
+  const bool is_Gaussian_curvature_selected,
+  const bool is_principal_curvatures_and_directions_selected,
+  const VPM vpm,
+  const VNM vnm
+)
+{
+  typedef typename boost::graph_traits<PolygonMesh>::face_descriptor Face_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor Vertex_descriptor;
```

Comment:
@sloriot do you confirm that it should be lowercase?

Reply:
No need to review that file, it will be remove as it comes from another branch. If you want to be strict it should be uppercase, but I'd use the lower case like a shortcut.